### PR TITLE
[MLX backend] Support for MLX backend across `layers` tests

### PR DESCRIPTION
**Pull request by acsweet**
## Description of the change
This PR is the first (of more to come) to add support for the MLX backend of Keras to work with Keras Hub.
The focus of this PR is tests in `keras_hub/src/layers`, with the following notes:
- Quantization is not yet implemented for the MLX backend
- MLX does not currently support float8
- MLX does not support matmul with integer data types
- masking is handled separately from a `_keras_mask` attribute, as it cannot be added to MLX `array` objects

Any feedback on changes and implementations here would be appreciated. Will modify as needed! 

Note that I have not included MLX in the GitHub CI yet. There are still a few issues failing tests on the Linux environment, so might need to wait until MLX is available via Keras nightly, also open to any other suggestsions.

## Checklist

- [x] I have added all the necessary unit tests for my change.
- [x] I have verified that my change does not break existing code and works with all backends (TensorFlow, JAX, and PyTorch).
- [x] My PR is based on the latest changes of the main branch (if unsure, rebase the code).
- [x] I have followed the Keras Hub [Model contribution guidelines](https://github.com/keras-team/keras-hub/blob/master/CONTRIBUTING_MODELS.md) in making these changes.
- [x] I have followed the Keras Hub [API design guidelines](https://github.com/keras-team/keras-hub/blob/master/API_DESIGN_GUIDE.md) in making these changes.
- [x] I have signed the [Contributor License Agreement](https://cla.developers.google.com/about).

### CODE REVIEW

**Comment by JyotinderSingh** [comment]
Thank you for this PR! The CLA error is coming because of incorrect rebasing/merging of this change with the master. As is visible from the commit list of the PR, there are lots of unrelated changes mentioned.

This usually happens due to erroneous merge workflows. You would ideally want to rebase/merge your branch with the latest master changes, and then **force push** those changes to your branch.

**Comment by acsweet** [comment]
Thank you @JyotinderSingh. I'm going to abandon this PR and open a new one. 

I think I missed something with git here.
